query-planner-js/src/buildPlan.ts

```diff
@@ -3821,24 +3816,39 @@ function extractDeferFromOperation({
   };
 }
 
-function addTypenameFieldForAbstractTypes(selectionSet: SelectionSet) {
-  for (const selection of selectionSet.selections()) {
-    if (selection.kind == 'FieldSelection') {
-      const fieldBaseType = baseType(selection.field.definition.type!);
-      if (isAbstractType(fieldBaseType)) {
-        selection.selectionSet!.add(new FieldSelection(new Field(fieldBaseType.typenameField()!)));
-      }
-      if (selection.selectionSet) {
-        addTypenameFieldForAbstractTypes(selection.selectionSet);
+function subselectionTypeIfAbstract(selection: Selection): AbstractType | undefined {
+  if (selection.kind === 'FieldSelection') {
+    const fieldBaseType = baseType(selection.element.definition.type!);
+    return isAbstractType(fieldBaseType) ? fieldBaseType : undefined;
+  } else {
+    const conditionType = selection.element.typeCondition;
+    return conditionType && isAbstractType(conditionType) ? conditionType : undefined;
+  }
+}
+
+function addTypenameFieldForAbstractTypes(selectionSet: SelectionSet, parentTypeIfAbstact?: AbstractType): SelectionSet {
```

Comment:
Misspell in the `parentTypeIfAbstact` symbol

Reply:
Another good catch, thanks.